From b05bac62fcb6f09241dd8f09170afc56f0db205d Mon Sep 17 00:00:00 2001
From: Bharat Bhushan <bharat.bhushan@freescale.com>
Date: Mon, 15 Apr 2013 09:12:13 +0530
Subject: [PATCH] KVM:PPC: use guest set WIMGE tlb attributes

commit b05bac62fcb6f09241dd8f09170afc56f0db205d git://git.freescale.com/ppc/sdk/linux.git

The guest will need to create cache-inhibited mappings for
direct device I/O.

Note that this means that the guest can create architecturally
undefined situations by mapping the same memory both cached and
uncached (either by itself with multiple mappings, or relative
to a host mapping).

This can be solved, but requires tracking either which physical address
/regions should be cacheable (similar to what /dev/mem does now), or
how each physical page has already been mapped.

Signed-off-by: Bharat Bhushan <bharat.bhushan@freescale.com>
Change-Id: I97e3f099e11631b3b9a81b4e5205d80624b6f730
Reviewed-on: http://git.am.freescale.net:8181/1486
Reviewed-by: Yoder Stuart-B08248 <stuart.yoder@freescale.com>
Reviewed-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
Tested-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
Signed-off-by: Vu Tran <vu.tran@windriver.com>

diff --git a/arch/powerpc/kvm/e500.h b/arch/powerpc/kvm/e500.h
index c2e5e98..6857385 100644
--- a/arch/powerpc/kvm/e500.h
+++ b/arch/powerpc/kvm/e500.h
@@ -117,7 +117,7 @@ static inline struct kvmppc_vcpu_e500 *to_e500(struct kvm_vcpu *vcpu)
 #define E500_TLB_USER_PERM_MASK (MAS3_UX|MAS3_UR|MAS3_UW)
 #define E500_TLB_SUPER_PERM_MASK (MAS3_SX|MAS3_SR|MAS3_SW)
 #define MAS2_ATTRIB_MASK \
-	  (MAS2_X0 | MAS2_X1)
+	(MAS2_X0 | MAS2_X1 | MAS2_W | MAS2_I | MAS2_M | MAS2_G | MAS2_E)
 #define MAS3_ATTRIB_MASK \
 	  (MAS3_U0 | MAS3_U1 | MAS3_U2 | MAS3_U3 \
 	   | E500_TLB_USER_PERM_MASK | E500_TLB_SUPER_PERM_MASK)
-- 
1.9.1

